test/position-managers/FeeCollection.t.sol

```diff
@@ -136,6 +137,107 @@ contract FeeCollectionTest is Test, PosmTestSetup, LiquidityFuzzers {
         assertEq(manager.balanceOf(address(lpm), currency1.toId()), 0);
     }
 
+    function test_collect_donate() public {
+        LiquidityRange memory range = LiquidityRange({poolKey: key, tickLower: -120, tickUpper: 120});
+        mint(range, 10e18, address(this), ZERO_BYTES);
+        uint256 tokenId = lpm.nextTokenId() - 1;
+
+        // donate to generate fee revenue
+        uint256 feeRevenue = 1e18;
+        donateRouter.donate(key, feeRevenue, feeRevenue, ZERO_BYTES);
+
+        BalanceDelta expectedFees = IPositionManager(address(lpm)).getFeesOwed(manager, tokenId);
+
+        // collect fees
+        uint256 balance0Before = currency0.balanceOfSelf();
+        uint256 balance1Before = currency1.balanceOfSelf();
+        BalanceDelta delta = collect(tokenId, ZERO_BYTES);
+
+        assertApproxEqAbs(uint256(int256(delta.amount0())), feeRevenue, 1 wei);
+        assertApproxEqAbs(uint256(int256(delta.amount1())), feeRevenue, 1 wei);
+        assertEq(delta.amount0(), expectedFees.amount0());
+        assertEq(delta.amount1(), expectedFees.amount1());
+
+        assertEq(uint256(int256(delta.amount0())), currency0.balanceOfSelf() - balance0Before);
+        assertEq(uint256(int256(delta.amount1())), currency1.balanceOfSelf() - balance1Before);
+    }
+
+    function test_collect_donate_sameRange() public {
+        // alice and bob create liquidity on the same range [-120, 120]
+        LiquidityRange memory range = LiquidityRange({poolKey: key, tickLower: -120, tickUpper: 120});
+
+        // alice provisions 3x the amount of liquidity as bob
+        uint256 liquidityAlice = 3000e18;
+        uint256 liquidityBob = 1000e18;
+
+        vm.startPrank(alice);
+        mint(range, liquidityAlice, alice, ZERO_BYTES);
+        vm.stopPrank();
+        uint256 tokenIdAlice = lpm.nextTokenId() - 1;
+
+        vm.startPrank(bob);
+        mint(range, liquidityBob, bob, ZERO_BYTES);
+        vm.stopPrank();
+        uint256 tokenIdBob = lpm.nextTokenId() - 1;
+
+        // donate to generate fee revenue
+        uint256 feeRevenue0 = 1e18;
+        uint256 feeRevenue1 = 0.1e18;
+        donateRouter.donate(key, feeRevenue0, feeRevenue1, ZERO_BYTES);
+
+        {
+            // alice collects her share
+            BalanceDelta expectedFeesAlice = IPositionManager(address(lpm)).getFeesOwed(manager, tokenIdAlice);
+            assertApproxEqAbs(
+                uint128(expectedFeesAlice.amount0()),
+                feeRevenue0.mulDivDown(liquidityAlice, liquidityAlice + liquidityBob),
+                1 wei
+            );
+            assertApproxEqAbs(
+                uint128(expectedFeesAlice.amount1()),
+                feeRevenue1.mulDivDown(liquidityAlice, liquidityAlice + liquidityBob),
+                1 wei
+            );
```

Comment:
these are just asserts checking that expectedFees is calculated correctly? maybe these kinds of tests should live in a unit test for getFeesOwed?

would be cool to fuzz some of this

Reply:
keeping the assertion since its asserting the size/magnitude of collected fees is equal to the calculated fees and the actual fees transferred (_delta_)

but added `test_fuzz_getFeesOwed_donate` which fuzzes donation revenue against getFeesOwed. there's some imprecision with getFeesOwed 😅